packages/lib-classifier/src/store/SubjectStore/SubjectStore.js

```diff
@@ -120,29 +120,11 @@ const SubjectStore = types
       next(call)
     }
 
-    function _onWorkflowChange() {
-      const workflow = tryReference(() => getRoot(self).workflows.active)
-      const subjectSet = tryReference(() => workflow?.subjectSet)
-      if (workflow || subjectSet) {
-        self.reset()
-        self.populateQueue(workflow.selectedSubjects)
-      }
```

Comment:
It would be nice if we could keep this, but modify it so that it doesn't run if we load in an existing subject store from a snapshot for a prioritised workflow. That would remove a bunch of extra conditional code from the classifier, which now manually resets the queue when the workflow or subject set or initial subject changes.

Reply:
There's two cases where we'd want to reset and repopulate the queue:
1. There's a workflow (and subject set for grouped workflows) but `self.resources` is empty, so we need to load subjects.
2. `self.resources` is not empty, but the subjects are not valid for the current workflow (or set.) This is the case that's impossible to validate with the current model.